sema/commons/j2/j2_functions.py

```diff
@@ -81,6 +81,15 @@ def xsd_format_date(content, quote, suffix):
         asdt = content
     return xsd_value(asdt.isoformat(), quote, "xsd:date")
 
+def xsd_format_gyear(content, quote, suffix):
+    # make rigid gYear
```

Comment:
check with the other formatters -- we typically support other data-type inputs as well

in this case it would make sense to support input in date and datetime format and forcefully convert it to YYYY format